### PR TITLE
Added change window size functionality including tests

### PR DESCRIPTION
**Pull request by Ad1B3nAr0ya**

### CODE REVIEW

File: src/commons/BrowserUtils.ts [hunk 1]
```diff
@@ -706,6 +708,36 @@ export namespace BrowserUtils {
...
+  ): void {
+      if(requestedWidth === undefined && requestedHeight === undefined ) {
+          Reporter.debug("No new window size requested, aborting window size action.");
+
+          return;
+      }
+    const size: Size = getWindowSize();
+    size.width = requestedWidth === undefined ? size.width : requestedWidth;
+    size.height = requestedHeight === undefined ? size.height : requestedHeight;
+    Reporter.debug(`size >> ${JSON.stringify(size)}`);
```
Comment (FelixZilber) [inline src/commons/BrowserUtils.ts]: make log more informative

File: src/commons/BrowserUtils.ts [hunk 2]
```diff
@@ -706,6 +708,36 @@ export namespace BrowserUtils {
...
+    browser.windowHandleSize(size);
+  }
+
+  /**
+   * Get current browser window size returns Size obj {width : number, height : number}
+   */
+  export function getWindowSize(): Size {
```
Comment (FelixZilber) [inline src/commons/BrowserUtils.ts]: incorrect function name - getWindowSize returns viewport size. it's not the same.
+ add Reporter.debug message

you might try to use:
window.innerWidth
window.innerHeight

File: test/specs/WindowSizeSpec.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+// tslint:disable-next-line:no-http-string
+const sampleAppUrl: string = 'http://127.0.0.1:8000/';
+
+/**
+ * wdio-allure-ts Change window size and get window size (setWindowSize, getWindowSize)
+ */
+describe('WindowSizeSpec of BrowserUtils Tests', () => {
+    it('Validate get window size default', () => {
+        BrowserUtils.navigateToUrl(sampleAppUrl);
+        const size : Size = BrowserUtils.getWindowSize();
```
Comment (FelixZilber) [inline test/specs/WindowSizeSpec.ts]: window.innerWidth
window.innerHeight
are JS call for getting the dimensions for comparison

File: src/commons/BrowserUtils.ts [hunk 4]
```diff
@@ -706,6 +708,36 @@ export namespace BrowserUtils {
...
 
+  /**
+   * Change size of browser window
+   *    If only one parameter is provided the second will stay as is.
+   *    If no parameter provided there will be no change.
+   * @param requestedWidth - Width (px)
+   * @param requestedHeight - Height (px)
+   */
+  export function setWindowSize(
+    requestedWidth?: number,
```
Comment (FelixZilber) [inline src/commons/BrowserUtils.ts]: need better api here.
maybe something like:  setWindwoSize({width:val, height:val})
this way you want see calls like:         BrowserUtils.setWindowSize(undefined, 768);
the undefined is very strange

File: test/wdio.conf.js [hunk 5]
```diff
@@ -11,16 +11,16 @@ exports.config = {
...
-      browserName: "firefox",
-      "moz:firefoxOptions": {
-        args: ["--headless"]
-      }
-    }
+        args: [ "--incognito"]
+      },
+     },
+    // {
+    //   maxInstances: 1,
```
Comment (FelixZilber) [inline test/wdio.conf.js]: revert

File: src/commons/BrowserUtils.ts (hunk 1 above)
Comment (Ad1B3nAr0ya) [reply src/commons/BrowserUtils.ts]: done

File: src/commons/BrowserUtils.ts (hunk 2 above)
Comment (Ad1B3nAr0ya) [reply src/commons/BrowserUtils.ts]: done

File: test/specs/WindowSizeSpec.ts (hunk 3 above)
Comment (Ad1B3nAr0ya) [reply test/specs/WindowSizeSpec.ts]: Runing script doesn't work as expected.

File: src/commons/BrowserUtils.ts (hunk 4 above)
Comment (Ad1B3nAr0ya) [reply src/commons/BrowserUtils.ts]: done

File: test/specs/WindowSizeSpec.ts (hunk 3 above)
Comment (Ad1B3nAr0ya) [reply test/specs/WindowSizeSpec.ts]: done

File: test/specs/ViewportSizeSpec.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+const sampleAppUrl: string = 'http://127.0.0.1:8000/';
+
+/**
+ * wdio-allure-ts Change window size and get window size (setViewportSize, getViewportSize)
+ */
+describe('ViewportSizeSpec of BrowserUtils Tests', () => {
+
+    it('Validate window size after size change to 1024x768', () => {
+        BrowserUtils.navigateToUrl(sampleAppUrl);
+        //BrowserUtils.setWindowSize({width:1024, height:768});
```
[the remark this answers is not in the document]
Comment (Ad1B3nAr0ya) [reply test/specs/ViewportSizeSpec.ts]: done

File: test/specs/ViewportSizeSpec.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        BrowserUtils.setViewportSize({ width: 1024, height: 768 });
+        browser.pause(1000);
```
[the remark this answers is not in the document]
Comment (Ad1B3nAr0ya) [reply test/specs/ViewportSizeSpec.ts]: removed

File: test/specs/ViewportSizeSpec.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        assert.equal(size.width, 1024, 'window width');
+        assert.equal(size.height, 768, 'window height');
+    });
+
+
+    it('Validate window size after size change to height = 768', () => {
+        BrowserUtils.navigateToUrl(sampleAppUrl);
+        BrowserUtils.setViewportSize({width: undefined, height: 768});
+        const size : Size = BrowserUtils.getViewportSize();
+        assert.equal(size.width, 1024, 'window width');
```
[the remark this answers is not in the document]
Comment (Ad1B3nAr0ya) [reply test/specs/ViewportSizeSpec.ts]: done

**Pull request merged**